collabora_online.module

```diff
@@ -114,3 +118,40 @@ function collabora_online_entity_operation(EntityInterface $entity) {
 
     return $entries;
 }
+
+/**
+ * Implements hook_ENTITY_TYPE_access() for 'media'.
+ *
+ * Checks access for the new media operations provided by this module.
+ */
+function collabora_online_media_access(
+    MediaInterface $media,
+    string $operation,
+    AccountInterface $account,
+): AccessResultInterface {
+    // @todo Should it make a difference whether a media is published or not?
+    // @todo Should a user be allowed to preview/edit a media in Collabora even
```

Comment:
These points are TBD.

Reply:
For regular media access in core:
- With 'administer media' you can do anything.
- (Generic "update any media" and "update media" and others are being faded out, I don't mention them below, see https://www.drupal.org/project/drupal/issues/2925459.)
- (Create and delete are not relevant for us, I don't mention them below.)
- Otherwise:
  - To view published media, it needs "view media" permission.
  - To view own unpublished media, it needs "view own unpublished media" permission to view.
  - To update any media (published or not), you need to:
    -  "edit any <type> media", OR.
    - "edit own <type> media", and be the owner.

Reply:
This actually reveals a lack in Drupal core.
See https://www.drupal.org/project/drupal/issues/2936652
Currently there are permissions to "edit any media" which also applies to unpublished, and does not require ownership.
But to view the same media, you need to be the owner or have "administer media".

Reply:
I am removing these `@todo` after talking with my colleagues.
In the current iteration, we make no distinction of published vs not published.
Also we treat these operations as isolated from other media operations.
Instead, we could add description to the permission itself.